src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -26,45 +26,61 @@ namespace pandaproxy::schema_registry {
 using server = ctx_server<service>;
 
 template<typename Handler>
-auto wrap(ss::gate& g, Handler h) {
-    return [&g, h{std::move(h)}](
+auto wrap(ss::gate& g, one_shot& es, Handler h) {
+    return [&g, &es, h{std::move(h)}](
              server::request_t rq,
              server::reply_t rp) -> ss::future<server::reply_t> {
+        co_await es();
         auto guard = gate_guard(g);
         co_return co_await h(std::move(rq), std::move(rp));
     };
 }
 
-server::routes_t get_schema_registry_routes(ss::gate& gate) {
+server::routes_t get_schema_registry_routes(ss::gate& gate, one_shot& es) {
```

Comment:
> my main question about this is: at some point the schema registry has to be created. why not perform initialization before its created and transfer control of the initialization state into the registry? would that avoid the need to have a check that occurs on every request, but is only true once for the lifetime? I think this approach is sound, but it does seem like there must be a cleaner solution.

That's a good idea. Might require a fair bit of refactoring and lifetime changes, but sounds like it might be cleaner.